gtk4/src/event_controller.rs

```diff
@@ -0,0 +1,24 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+#[cfg(feature = "v4_8")]
+use glib::translate::*;
+
+#[cfg(feature = "v4_8")]
+use crate::ffi;
+use crate::{prelude::*, EventController};
+
+pub trait EventControllerExtManual: IsA<EventController> + 'static {
+    #[cfg(feature = "v4_8")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "v4_8")))]
+    #[doc(alias = "gtk_event_controller_set_static_name")]
+    fn set_static_name(&self, name: Option<&'static str>) {
```

Comment:
```suggestion
    fn set_static_name(&self, name: Option<&'static glib::GStr>) {
```

it needs to be a NUL-terminated string, and `&str` is not

Reply:
```suggestion
    fn set_static_name(&self, name: impl Into<Option<&'static glib::GStr>>) {
```

if you want it to be a bit more generic

Reply:
https://gtk-rs.org/gtk-rs-core/stable/latest/docs/src/glib/translate.rs.html#558-590

That's already covered by the function `to_glib_none`

Reply:
It's better to have API simplicity and accept a `&str`.

Reply:
This doesn't fix anything though. The temporary string created by `to_glib_none()` is freed and then you have a dangling pointer inside GTK.

Reply:
Do you think we should ***remove*** this function and encourage people to use `set_name`?

The reason being that API users will have to convert the string into a `CStr` then into a `GStr`, resulting in very verbose code.

`GStr` doesn't have `From` traits, but it has `TryFrom<&'a CStr>`, so even if the function signature is `fn set_static_name(name: impl Into<Option<&'static glib::GStr>>)` it won't work.

An example of calling `set_static_name`:
```rust
let controller: gtk::EventController = ...;
controller.set_static_name(Some(c"Hello".try_into().unwrap()));
```

Doing so will represent a breaking change

Reply:
Nobody has reported this error since 3 years ago, so I think is less of a problem to break the API.

https://github.com/gtk-rs/gtk4-rs/blame/main/gtk4/src/auto/event_controller.rs#L148

Reply:
You can do `glib::gstr!("blablabla")` to get a `&'static glib::GStr`

Reply:
Ok, I'm gonna add that to the PR and re-open it